### PR TITLE
Upgrade to v2.42.0 of the AzureRM Terraform Provider

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #808

### CODE REVIEW

File: sdk/dotnet/Network/NetworkSecurityRule.cs [hunk 1]
```diff
@@ -247,11 +247,17 @@ public InputList<string> DestinationAddressPrefixes
...
 
+        [Input("destinationApplicationSecurityGroupIds")]
+        private InputList<string>? _destinationApplicationSecurityGroupIds;
+
         /// <summary>
         /// A List of destination Application Security Group ID's
         /// </summary>
-        [Input("destinationApplicationSecurityGroupIds")]
-        public Input<string>? DestinationApplicationSecurityGroupIds { get; set; }
+        public InputList<string> DestinationApplicationSecurityGroupIds
```
Comment (mikhailshilkov) [inline sdk/dotnet/Network/NetworkSecurityRule.cs:256]: Some breaking changes here and below. Let's triage.

Comment (stack72) [reply sdk/dotnet/Network/NetworkSecurityRule.cs:256]: @mikhailshilkov this is an actual change in the provider

https://github.com/terraform-providers/terraform-provider-azurerm/pull/9884/files

Comment (mikhailshilkov) [reply sdk/dotnet/Network/NetworkSecurityRule.cs:256]: I guess we just changelog this and go ahead?

**Review by mikhailshilkov — APPROVED** [review]
LGTM, thanks for fixing!

**Pull request merged**